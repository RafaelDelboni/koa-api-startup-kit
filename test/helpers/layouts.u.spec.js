process.env.APP_NAME = 'your-super-app-name'

const mockReadFileSync = jest.fn(() => '<p><b>{{appName}}</b>{{user.name}}</p>')
const mockReaddirSync = jest.fn(() => ['file-one.hbs', 'file-two.hbs'])

jest.mock('fs', () => {
  return {
    readFileSync: mockReadFileSync,
    readdirSync: mockReaddirSync
  }
})

const template = require('../../src/helpers/template')

describe('unit', () => {
  describe('helpers', () => {
    describe('layouts', () => {
      describe('render', () => {
        it('should render html', async function () {
          const html = template('file-two', {
            user: {
              name: 'test'
            }
          })
          expect(html).toMatchSnapshot()
        })
        it('should not render html', async function () {
          expect.assertions(1)
          try {
            template('file-that-not-exists', {
              user: {
                name: 'test'
              }
            })
          } catch (error) {
            expect(error.message).toEqual(
              'Template file-that-not-exists not found'
            )
          }
        })
      })
    })
  })
})
