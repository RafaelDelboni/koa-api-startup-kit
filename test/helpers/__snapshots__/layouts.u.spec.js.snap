// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit helpers layouts render should render html 1`] = `"<p><b>your-super-app-name</b>test</p>"`;
